lib/waterfoul/ppu.rb

```diff
@@ -37,19 +37,19 @@ def step(cycles = 1)
       @modeclock += cycles
       @auxillary_modeclock += cycles
 
-      if IO::LCDControl.screen_enabled? && @screen_enabled
-        case @mode
-        when H_BLANK_STATE
-          hblank if @modeclock >= H_BLANK_TIME
-        when V_BLANK_STATE
-          vblank
-        when OAM_READ_STATE
-          oam if @modeclock >= OAM_SCANLINE_TIME
-        when VMRAM_READ_STATE
-          vram if @modeclock >= VRAM_SCANLINE_TIME
-        end
-      else
-        if IO::LCDControl.screen_enabled?
+      if IO::LCDControl.screen_enabled?
```

Comment:
It avoids calling `IO::LCDControl.screen_enabled?` and slightly simplifies the control flow.